resolver/dns/dns_resolver.go

```diff
@@ -269,11 +269,16 @@ func (d *dnsResolver) watcher() {
 
 		// Sleep to prevent excessive re-resolutions. Incoming resolution requests
 		// will be queued in d.rn.
-		t := time.NewTimer(minDNSResRate)
+		var t *time.Timer
```

Comment:
@dfawley 

> Perhaps you meant to define the variable outside the for loop?

Yes. 

We should use a timer global and reset it instead of every time malloc a new timer.

Reply:
> We should use a timer global and reset it instead of every time malloc a new timer.

Yes, but the benefit is minimal (one allocation every 30 minutes), with the cost of increased code complexity.  But, we will be removing this polling entirely, in either this or the next release, so I don't think it's worth making changes here right now.  Thank you for the suggestions.